tests/flytekit/unit/core/test_flyte_file.py

```diff
@@ -433,3 +435,19 @@ def wf(path: str) -> os.PathLike:
         return t2(ff=n1)
 
     assert flyte_tmp_dir in wf(path="s3://somewhere").path
+
+
+def test_flyte_file_annotated_hashmethod():
+    def calc_hash(ff: FlyteFile) -> str:
+        return str(ff.path)
+
+    @task
+    def t1(path: str) -> Annotated[FlyteFile, HashMethod(calc_hash)]:
+        return FlyteFile(path)
+
+    @workflow
+    def wf(path: str) -> None:
+        t1(path=path)
+
+    with tempfile.NamedTemporaryFile() as f:
```

Comment:
`NamedTemporaryFile` objects are weird on [windows](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile) (essentially they can't be opened twice). Specifying `delete=False` should fix the issue.

Reply:
Sorry for the slow response, I was OOO last week.

I noticed the `local_dummy_file` fixture in the test module and switched to using that instead in c74dca15.